config/config.go

```diff
@@ -97,11 +108,20 @@ func Parse(in *MountParams) (*MountConfig, error) {
 		return nil, fmt.Errorf("failed to unmarshal secrets: %v", err)
 	}
 	if _, ok := secret["key.json"]; ok {
-		creds, err := google.CredentialsFromJSON(context.Background(), []byte(secret["key.json"]), "https://www.googleapis.com/auth/cloud-platform")
-		if err != nil {
-			return nil, fmt.Errorf("unable to generate credentials from key.json: %w", err)
+		out.AuthNodePublishSecret = true
+		out.AuthKubeSecret = []byte(secret["key.json"])
+	}
+
+	if attrib["auth"] == "self-adc" {
+		if out.AuthNodePublishSecret {
+			klog.InfoS("attempting to set both NodePublish and ADC auth. Using NodePublish.", podInfo)
```

Comment:
Should this be an error to help users catch misconfigurations instead of mistakenly thinking it's one or the other?

Reply:
sgtm, updated